anxcloud/testutils/recorder/vm_recorder.go

```diff
@@ -0,0 +1,79 @@
+package recorder
+
+import (
+	"context"
+	"errors"
+	"go.anx.io/go-anxcloud/pkg/api"
+	"go.anx.io/go-anxcloud/pkg/client"
+	"go.anx.io/go-anxcloud/pkg/vsphere/provisioning/vm"
+	"go.anx.io/go-anxcloud/pkg/vsphere/search"
+	"net/http"
+)
+
+type vmCleanUpFunc func(ctx context.Context) error
+type VMRecoder struct {
+	handlers []vmCleanUpFunc
+	client   api.API
```

Comment:
is this uses for anything? I see some clients being created, maybe store that here instead of a generic client, that cannot do ADC, yet?